app-frontend/src/app/index.routes.js

```diff
@@ -291,70 +292,69 @@ function settingsStates($stateProvider) {
         });
 }
 
-function marketStates($stateProvider) {
-    $stateProvider
-        .state('market', {
-            title: 'Tools',
-            parent: 'root',
-            url: '/market',
-            templateUrl: marketTpl,
-            controller: 'MarketController',
-            controllerAs: '$ctrl',
-            abstract: true
-        })
-        .state('market.search', {
-            title: 'Tool Search',
-            url: '/search?:page?:query?toolcategory&tooltag',
-            templateUrl: marketSearchTpl,
-            controller: 'MarketSearchController',
-            controllerAs: '$ctrl'
-        })
-        .state('market.tool', {
-            title: 'Tool Details',
-            url: '/tool/:id',
-            params: {
-                modelData: null
-            },
-            templateUrl: marketToolTpl,
-            controller: 'MarketToolController',
-            controllerAs: '$ctrl'
-        });
-}
-
 function labStates($stateProvider) {
     $stateProvider
         .state('lab', {
-            title: 'Tool Lab',
+            title: 'Lab',
+            template: '<ui-view></ui-view>',
             parent: 'root',
-            url: '/lab/:toolid',
+            url: '/lab',
+            redirectTo: 'lab.browse'
+        })
+        .state('lab.build', {
+            title: 'Build a Tool',
+            url: '/build/:toolid',
+            parent: 'lab',
             templateUrl: labTpl,
             controller: 'LabController',
             controllerAs: '$ctrl',
-            abstract: true
+            redirectTo: 'lab.build.run'
+        })
+        .state('lab.build.run', {
+            title: 'Run',
+            url: '/run/:runid?',
+            parent: 'lab.build',
+            views: {
+                'navmenu@root': {
+                    templateUrl: labNavbarTpl,
+                    controller: 'LabNavbarController',
+                    controllerAs: '$ctrl'
+                },
+                '': {
+                    templateUrl: labRunTpl,
+                    controller: 'LabRunController',
+                    controllerAs: '$ctrl'
+                }
+            }
         })
         .state('lab.edit', {
```

Comment:
I think we can get rid of this (and the labEdit stuff) since it will all be a part of lab.build